src/service/submit.rs

```diff
@@ -55,3 +55,26 @@ pub fn submission_list(
     })()
     .context("Failed querying submissions from the server")
 }
+
+pub fn nth_latest_submission_info(
+    res: &mut Resources<impl RP>,
+    scope: &Scope,
+    task_id: &str,
+    n: u64,
+) -> Result<SubmissionInfo> {
+    (|| {
+        let submissions = submission_list(res, scope, task_id)?;
+        let idx = submissions
+            .len()
+            .checked_sub(1 + n as usize)
+            .ok_or_else(|| {
+                anyhow!(format!(
+                    "The nth latest submission doesn't exist for n = {}",
+                    n
+                ))
```

Comment:
`anyhow!` is a macro because it permits formatting directly:
```suggestion
                anyhow!(
                    "The nth latest submission doesn't exist for n = {}",
                    n
                )
```